### PR TITLE
Spiral sync

### PR DESCRIPTION
**Pull request by misternebula**

### CODE REVIEW

**Comment by misternebula** [comment]
also includes some namespace changes - sorry 🥴

File: README.md [hunk 1]
```diff
@@ -37,6 +38,13 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
 ## Frequently Asked Questions
 
+### Requirements
+- Latest version of OWML.
+- Latest version of Mod Manager. (If using)
+- Latest version of Outer Wilds. (Epic version preferred, as Steam version is untestable.)
+- Fast and stable internet connection, upload and download.
+- Should be able to run on minimum Outer Wilds system requirements.
+
```
Comment (amazingalek) [inline README.md:47]: and open ports?

File: README.md [hunk 2]
```diff
@@ -47,6 +55,13 @@ QSB does not operate on a strict server-client relationship. Each player's game
 - Player ships
 - Player probes
 - NPC heads in conversations
+- Ship log
+- Discovering frequencies / signals
+- Nomai text (Spirals + Computers)
+
+QSB also changes some mechanics of the base game, to better fit a multiplayer experience. These include :
+- Adding dialogue boxes above NPC and player heads, so other players can "listen in" on conversations.
+- Quantum objects check observations from all players and all player probes.
```
Comment (amazingalek) [inline README.md:65]: and respawn instead of normal death

File: QSB/TranslationSync/WorldObjects/QSBWallText.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+		public IEnumerable<int> GetTranslatedIds()
+		{
+			var dict = AttachedObject.GetValue<Dictionary<int, OWTreeNode<NomaiTextLine>>>("_idToNodeDict");
+			foreach (var key in dict.Keys)
+			{
+				if (AttachedObject.IsTranslated(key))
+				{
+					yield return key;
+				}
+			}
```
Comment (amazingalek) [inline QSB/TranslationSync/WorldObjects/QSBWallText.cs:33]: `return dict.Keys.Where(key => AttachedObject.IsTranslated(key));` ?

File: QSB/TranslationSync/Events/SetAsTranslatedEvent.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+			else if (message.TextType == NomaiTextType.Computer)
+			{
+				var obj = QSBWorldSync.GetWorldObject<QSBComputer>(message.ObjectId);
+				obj.HandleSetAsTranslated(message.TextId);
+			}
+			else if (message.TextType == NomaiTextType.VesselComputer)
+			{
+				var obj = QSBWorldSync.GetWorldObject<QSBVesselComputer>(message.ObjectId);
+				obj.HandleSetAsTranslated(message.TextId);
+			}
```
Comment (amazingalek) [inline QSB/TranslationSync/Events/SetAsTranslatedEvent.cs:44]: if QSBWallText, QSBComputer and QSBVesselComputer inherits from common class/interface, you could extract some of this out of the if-else:

var obj = GetObj(...); // use if-else or switch
obj.HandleSetAsTranslated(message.TextId);

not a big deal tho, it's fine as-is ;)

**Pull request merged**